src/mixins/keyboard-shortcut-mixin.ts

```diff
@@ -1,26 +1,69 @@
 import { LitElement } from "lit-element";
 import { Constructor } from "../types";
 
+type RegisteredShortcuts = {
+  [key in HAKeyboardShortcut]?: () => void;
+};
+
+export enum HAKeyboardShortcut {
+  CTRL_S = "CtrlS",
+  CTRL_P = "CtrlP",
+  CTRL_SHIFT_P = "CtrlShiftP",
+}
+
+const isMac = /(Mac|iPhone|iPod|iPad)/i.test(navigator.platform);
+
 export const KeyboardShortcutMixin = <T extends Constructor<LitElement>>(
   superClass: T
 ) =>
   class extends superClass {
-    private _keydownEvent = (event: KeyboardEvent) => {
-      if ((event.ctrlKey || event.metaKey) && event.key === "s") {
-        event.preventDefault();
-        this.handleKeyboardSave();
-      }
-    };
-
-    public connectedCallback() {
-      super.connectedCallback();
-      this.addEventListener("keydown", this._keydownEvent);
-    }
+    private _registeredShortcuts: RegisteredShortcuts = {};
 
     public disconnectedCallback() {
       this.removeEventListener("keydown", this._keydownEvent);
       super.disconnectedCallback();
     }
 
-    protected handleKeyboardSave() {}
+    private _executeShortcut(
+      event: KeyboardEvent,
+      shortcut: HAKeyboardShortcut
+    ) {
+      if (Object.keys(this._registeredShortcuts).includes(shortcut)) {
+        event.preventDefault();
+        const shortcutAction = this._registeredShortcuts[shortcut];
+        if (shortcutAction) {
+          shortcutAction();
+        }
+      }
+    }
+
+    protected registerShortcut(
+      shortcut: HAKeyboardShortcut,
+      callback: () => void,
+      element: HTMLElement | Document = this
+    ) {
+      element.addEventListener("keydown", (event) => this._keydownEvent(event));
```

Comment:
We should not do this in `registerShortcut`, we would get multiple listeners if you register multiple shortcuts.